go/tools/builders/link.go

```diff
@@ -94,7 +105,7 @@ This will be an error in the future.`, pkgPath, label, conflictLabel)
 		depsSeen[pkgPath] = label
 
 		pkgSuffix := string(os.PathSeparator) + filepath.FromSlash(pkgPath) + ".a"
-		if !strings.HasSuffix(pkgFile, pkgSuffix) {
+		if !strings.HasSuffix(filepath.FromSlash(pkgFile), pkgSuffix) {
```

Comment:
Is this needed? It looks like this should come from `d.file.path` in link.bzl:78. I would expect that to have the correct separator, but I can't verify at the moment.

Reply:
Yes it is needed. It comes in with `/` on Windows, but `pkgSuffix` uses the OS specific separator.